homeassistant/components/rotel/media_player.py

```diff
@@ -0,0 +1,493 @@
+        """
+        Reconnect.
+
+        Don't call this more often than needed - it is asynchronous
+        and it doesn't check for activity in progress on the connection
+        so there could be a risk of a conflict.
+        """
+        await self.cancel_read_messages()
+
+        # Ignore any errors while closing the connection because
+        # the reason we'd be doing this would probably be due to some
+        # sort of issue with the existing connection anyway.
+        try:
+            self.close_connection()
+        # pylint: disable=broad-except
+        except Exception:
+            _LOGGER.error("Could not close connection", exc_info=True)
+            _LOGGER.warning("Ignoring error and attempting to reconnect")
+
+        # Set the state to OFF by default
+        # If the player is actually on then the state will be refreshed
+        # when the message reader restarts
+        self._state = STATE_OFF
+        self.async_schedule_update_ha_state()
+
+        await self.open_connection()
+
+        # N.B. this line can only be executed after
+        # async_add_entities has been called and completed
+        # because otherwise self.hass won't be available.
+        self.start_read_messages(self.hass)
+
+    async def cleanup(self):
+        """Close connection and stop message reader."""
+        _LOGGER.info("Cleaning up '%s'", self.name)
+        await self.cancel_read_messages()
+        self.close_connection()
+        _LOGGER.info("Finished cleaning up '%s'", self.name)
+
+    def set_source_lists(self, source_aliases):
+        """
+        Set list of sources that can be selected.
+
+        For sources that have an alias defined then let the user pick that
+        instead. To use an alias for a source, provide an entry in the
+        source_aliases map of the form <source>: <alias>
+        To suppress a source, provide an entry in the source_aliases
+        map of the form <source>: None
+        Any sources not suppressed and not aliased will remain as-is
+        Note that the amp could still be switched to one of the suppressed
+        sources using the remote control or front panel.  The
+        media player card seems to handle this gracefully enough.
+        Example of using source_aliases in configuration.yaml:
+
+        media_player:
+        - platform: rotel
+          device: /dev/ttyUSB0
+          source_aliases:
+            TAPE:
+            MULTI:
+            VIDEO 1: CATV
+            VIDEO 2: NMT
+            VIDEO 3: APPLE TV
+            VIDEO 4: FIRE TV
+            VIDEO 5: BLU RAY
+        """
+        _LOGGER.debug("source_aliases: %r", source_aliases)
+        sources_to_select = {}
+        aliased_sources = set()
+        if source_aliases is not None:
+            for source, alias in source_aliases.items():
+                if alias is None:
+                    aliased_sources.add(source)
+                else:
+                    aliased_sources.add(source)
```

Comment:
Should this be `alias`?  Otherwise this line is the same as the `if` condition and should just be moved outside the `if` block.